internal/juju/jaas.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package juju
+
+import (
+	"context"
+	"errors"
+
+	"github.com/canonical/jimm-go-sdk/v3/api"
+	"github.com/canonical/jimm-go-sdk/v3/api/params"
+	jujuapi "github.com/juju/juju/api"
+)
+
+type jaasClient struct {
+	SharedClient
+	getJaasApiClient func(jujuapi.Connection) JaasAPIClient
```

Comment:
nitpick: we could call this `newAPIClient`.. we know it's a JAAS api client as it is a method of the jaasClient

Reply:
It's at least in line with the naming style used for the other clients, but I see your point.